Include/internal/pycore_pymem.h

```diff
@@ -91,6 +95,18 @@ static inline int _PyMem_IsULongFreed(unsigned long value)
 #endif
 }
 
+static inline int
+_PyMem_Annotate_Mmap(void *addr, size_t size, const char *name)
```

Comment:
I always find it weird when we have an uppercase letter followed by its lowercase form (`Mm`). How about... `AnnotateMemoryMap` instead? that way, if Windows is ever going to support this kind of thing, we could add an implementation for it (and it wouldn't just for `mmap`).